Vue Component.sublime-syntax.yaml-macros

```diff
@@ -0,0 +1,428 @@
+%YAML 1.2
+%TAG ! tag:yaml-macros:vue_macros:YAMLMacros.lib.extend:
+---
+!extend
+_base: HTML.sublime-syntax
+name: Vue Component
+scope: text.html.vue
+hidden: false
+file_extensions: [vue]
+
+contexts: !merge
+  main: !prepend
+    - include: langs
+    - include: template-tag
+
+  tag-attributes: !prepend
+    - include: vue-directive
+
+  template-tag:
+    - match: '(?i)(<)(template)\b'
+      captures:
+        0: meta.tag.template.begin.html
+        1: punctuation.definition.tag.begin.html
+        2: entity.name.tag.template.html
+      push:
+        - match: (?i)(</)(template)(>)
+          captures:
+            0: meta.tag.template.end.html
+            1: punctuation.definition.tag.begin.html
+            2: entity.name.tag.template.html
+            3: punctuation.definition.tag.end.html
+          pop: true
+        - match: '>'
+          scope: meta.tag.template.begin.html punctuation.definition.tag.end.html
+          push: mustache-template
+        - match: ''
+          push:
+            - meta_scope: meta.tag.template.begin.html
+            - match: '(?=>)'
+              pop: true
+            - include: tag-attributes
+
+  mustache-template:
+    - match: (?=</template)
+      pop: true
+    - match: '{{'
+      scope: punctuation.definition.template.begin.html
+      embed: scope:source.js
+      escape: '}}'
+      escape_captures:
+        0: punctuation.definition.template.end.html
+    - include: main
+
+  vue-directive:
+    - match: \b(v-[\w-]+)\b
+      scope: entity.other.attribute-name.html
+      push: js-attribute-value
+    - match: \B(:|@)([\w-]+)\b
+      captures:
+        1: punctuation.definition.attribute.html
+        2: entity.other.attribute-name.html
+      push: js-attribute-value
+
+  js-attribute-value:
+    - match: '='
+      scope: punctuation.separator.key-value.html
+      set: js-string
+    - include: else-pop
+
+  js-string:
+    - match: '"'
+      scope: punctuation.definition.string.begin.html
+      embed: scope:source.js
+      escape: '"'
+      escape_captures:
+        0: punctuation.definition.string.end.html
+    - match: "'"
+      scope: punctuation.definition.string.begin.html
+      embed: scope:source.js
+      escape: "'"
+      escape_captures:
+        0: punctuation.definition.string.end.html
+    - include: else-pop
+
+  langs:
+    - match: (?:^\s+)?(<)((?:script))\b(?=[^>]*lang\s*=\s*(['"])coffee\1?)
```

Comment:
The backreference will trigger Sublime's old, slower regexp engine. This could be rewritten without it for improved performance.

Also, why match `(?:^\s+)?`? I don't think that this should be marked with the meta scope.

Reply:
Removed backreference. Not sure why that was even required. Seems to work without the backreference.

Removed the `(?:^\s+)?`. I'm not great with regexp and used some snippets from the old syntax where this came from 😅